### PR TITLE
[Refactor] Added unit test for call on hold feature

### PR DESCRIPTION
**Pull request by kevinyulu**
## Purpose

* Added unit test for call on hold feature

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

* Run unit test and see if there is any unit test failed.

## What to Check
Verify that the following are valid
* Check if there is anything to be added in the unit test for CallingMiddlewareTests, OnHoldOverlayViewModelTests, CallingServiceTests and CallingViewModelTests.
* The unit test for the above mentioned test classes must pass.

### CODE REVIEW

**Comment by palatter** [comment]
Can we test resume button?

**Comment by palatter** [comment]
> Can we test resume button?

Something like test_localVideoVideModel_toggleCameraSwitch_when_cameraStatusOn_then_shouldRequestCameraOnTriggered

**Review by AmyL219 — COMMENTED** [review]
We may need to add unit tests for `AudioSessionReducer`, [CallingMiddlewareHandler](https://github.com/Azure/communication-ui-library-ios/commit/090793e81511ae3025c479db32b1c44f0e846d3c#diff-7cf51eb35c41ba775e7bbed8d0e028181f7d003c9aa44aa8db2b5595d3fcfc56), and new added public functions inside the viewModels. https://github.com/Azure/communication-ui-library-ios/pull/229/files

**Comment by kevinyulu** [comment]
> > Can we test resume button?
> 
> Something like test_localVideoVideModel_toggleCameraSwitch_when_cameraStatusOn_then_shouldRequestCameraOnTriggered

I thought about testing the tap action as well , as the button tap is not driven by any state change in view model. it may not seem to be testable though.

**Comment by kevinyulu** [comment]
> We may need to add unit tests for `AudioSessionReducer`, [CallingMiddlewareHandler](https://github.com/Azure/communication-ui-library-ios/commit/090793e81511ae3025c479db32b1c44f0e846d3c#diff-7cf51eb35c41ba775e7bbed8d0e028181f7d003c9aa44aa8db2b5595d3fcfc56), and new added public functions inside the viewModels. https://github.com/Azure/communication-ui-library-ios/pull/229/files

@AmyL219  
I added the unit test for `AudioSessionReducer` and `CallingMiddlewareHandler`, please check if there is anything to be tested further.
For the changes inside view Model https://github.com/Azure/communication-ui-library-ios/pull/229/files, can I include this in another PR as this PR currently addresses the unit tests for call on hold

**Comment by palatter** [comment]
> > > Can we test resume button?
> > 
> > 
> > Something like test_localVideoVideModel_toggleCameraSwitch_when_cameraStatusOn_then_shouldRequestCameraOnTriggered
> 
> I thought about testing the tap action as well , as the button tap is not driven by any state change in view model. it may not seem to be testable though.

Maybe we could create a mock of primaryButtonViewModel, that the mock factory returns. Then have a method that on the mock that 'taps' the button, which executes the action given to it?

**Review by AmyL219 — DISMISSED** [review]

**Comment by kevinyulu** [comment]
I will create a ticket to address the unit test work to be added inside view Model https://github.com/Azure/communication-ui-library-ios/pull/229/files, can I include this in another PR as this PR currently addresses the unit tests for call on hold

**Comment by AmyL219** [comment]
> I will create a ticket to address the unit test work to be added inside view Model https://github.com/Azure/communication-ui-library-ios/pull/229/files.

Sounds good. Most of the function inside viewModel should be private level. Be free to update the function to be private if I miss that

**Review by palatter — DISMISSED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICallingTests/Presentation/Calling/CallingViewModelTests.swift [hunk 1]
```diff
@@ -168,6 +168,21 @@ class CallingViewModelTests: XCTestCase {
         sut.receive(appState)
         wait(for: [expectation], timeout: timeout)
     }
+
+    func test_callingViewModel_receive_when_callingStateStatusIsLocalHold_then_isOnHoldOverlayDisplayed_shouldBeTrue() {
+
+        let sut = makeSUT()
+        let appState = AppState(callingState: CallingState(status: .localHold))
+        sut.receive(appState)
+        XCTAssertEqual(sut.onHoldOverlayViewModel.isDisplayed, true)
```
Comment (jimchou-dev) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICallingTests/Presentation/Calling/CallingViewModelTests.swift:177]: XCTAssertTrue

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICallingTests/Redux/Middleware/CallingMiddlewareHandlerTests.swift [hunk 2]
```diff
@@ -407,6 +407,30 @@ class CallingMiddlewareHandlerTests: XCTestCase {
...
+        }
+        callingMiddlewareHandler.holdCall(state: state, dispatch: getEmptyDispatch())
+        XCTAssertTrue(mockCallingService.holdCallCalled)
+    }
+
+    func test_callingMiddlewareHandler_resumeCall_then_resumeCallCalled() {
+        guard let state: AppState = getState(callingState: .localHold,
+                                       cameraStatus: .off,
+                                       cameraDeviceStatus: .front,
+                                             cameraPermission: .notAsked) as? AppState else {
```
Comment (jimchou-dev) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICallingTests/Redux/Middleware/CallingMiddlewareHandlerTests.swift:426]: nitpick: indentation

**Review by jimchou-dev — DISMISSED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICallingTests/Presentation/Calling/OnHoldOverlayViewModelTests.swift [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    var localizationProvider: LocalizationProviderMocking!
+    var factoryMocking: CompositeViewModelFactoryMocking!
+    var accessibilityProvider: AccessibilityProviderMocking!
+
+    override func setUp() {
+        super.setUp()
+        logger = LoggerMocking()
+        storeFactory = StoreFactoryMocking()
+        localizationProvider = LocalizationProviderMocking()
+        factoryMocking = CompositeViewModelFactoryMocking(logger: LoggerMocking(), store: storeFactory.store)
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICallingTests/Presentation/Calling/OnHoldOverlayViewModelTests.swift:22]: logger can be reused

**Review by vhuseinova-msft — DISMISSED** [review]

**Review by vhuseinova-msft — APPROVED** [review]

**Review by palatter — APPROVED** [review]
Looks good!

**Pull request merged**